test/commands-tests/compatibility-tests.js

```diff
@@ -0,0 +1,136 @@
+
+const {
+    exec,
+    spawn
+} = require('promisify-child-process');
+
+const constants = require('../constants.json');
+const testFolder = constants.compatibilityTestFolder;
+const cliCommands = constants.cliCommands;
+const cliCmdOptions = constants.cliCommandsOptions;
+
+const aeConfig = require('aeproject-config');
+const nodeConfig = aeConfig.nodeConfiguration;
+const compilerConfig = aeConfig.compilerConfiguration;
+
+let executeOptions = {
+    cwd: process.cwd() + testFolder
+};
+
+const compatibilityCmd = (options) => {
+    return new Promise((resolve, reject) => {
+        let cmd = `${ cliCommands.AEPROJECT } ${ cliCommands.COMPATIBILITY }`;
+        if (options && options.nodeVersion) {
+            cmd += ` ${ cliCmdOptions.NODE_VERSION } ${ options.nodeVersion }`;
+        }
+
+        if (options && options.compilerVersion) {
+            cmd += ` ${ cliCmdOptions.COMPILER_VERSION } ${ options.compilerVersion }`;
+        }
+
+        let result = '';
+        let log = '';
+
+        let temp = exec(cmd, executeOptions);
+        if (temp.stdout) {
+            temp.stdout.on('data', async (data) => {
+                let str = data.toString('utf8');
+                log += str;
+                if (str.trim() == 'Running tests...') {
+                    result = await exec(cliCommands.DOCKER_PS);
+                }
+            });
+        }
+
+        if (temp.stderr) {
+            temp.stderr.on('data', (data) => {
+                console.log('err:', data.toString('utf8'));
+            });
+        }
+
+        temp.on('exit', (code) => {
+            if (code !== 0) {
+                const msg = `Child process exited with code ${ code }`;
+                console.log(msg);
+                reject(msg)
+            }
+            
+            if (options && options.logs) {
+                resolve(log);
+            } else {
+                resolve(result);
+            }
+        });
+    });
+}
+
+describe('Compatibility tests', async function () {
+    let tempCWD = process.cwd();
+
+    before(async function () {
+        fs.ensureDirSync(`.${ testFolder }`);
+        await execute(constants.cliCommands.INIT, [], executeOptions);
+        process.chdir(executeOptions.cwd);
+    })
+
+    it('Docker images should be run with "latest" versions', async function () {
+        let result = await compatibilityCmd();
+
+        if (result && result.stdout) {
+            const isNodeLatestVersion = result.stdout.indexOf(`${ nodeConfig.dockerImage }:latest`) >= 0;
+            const isCompilerLatestVersion = result.stdout.indexOf(`${ compilerConfig.dockerImage }:latest`) >= 0;
+
+            assert.isOk(isNodeLatestVersion && isCompilerLatestVersion, 'Node is not running with latest version');
+            assert.isOk(isCompilerLatestVersion, 'Compiler is not running with latest version');
+        } else {
+            assert.isOk(false, 'Cannot get result of "docker ps" command');
```

Comment:
is this needed?

Reply:
yes, it's needed. I refactored it but if inside 'compatibilityCmd' exited code is different from 0, it will return an error message. In short, if there is no 'result.stdout', execution is always failed

Reply:
Sorry, my question was, do you need the else, if the statements in the firs place are not true ins't the test faield ?

Reply:
I refactored the code, 'if' statement was removed